### PR TITLE
[Remove Stream/Optional usage] LPS-170733 Remove usages of Optional on Method Return Value for module segments-asah-connector | part 2

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:tango

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
Force-Push is just to use ListUtil.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:tango

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(12/29 9:33)

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/context/contributor/SegmentsAsahRequestContextContributor.java [hunk 1]
```diff
@@ -67,17 +66,21 @@ private String _getSegmentsAnonymousUserId(
...
+					cookie.getName(), _AC_CLIENT_USER_ID_COOKIE_NAME)) {
+
+				String value = cookie.getValue();
+
+				if (value != null) {
+					return value;
+				}
+
+				break;
+			}
```
Comment (yuhai) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/context/contributor/SegmentsAsahRequestContextContributor.java:80]: @W2jx, break is not necessary.

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/messaging/InterestTermsChecker.java [hunk 2]
```diff
@@ -75,19 +75,15 @@ public void checkInterestTerms(long companyId, String userId) {
...
-		String[] terms = stream.flatMap(
-			topic -> {
-				List<Topic.TopicTerm> topicTerms = topic.getTerms();
-
-				return topicTerms.stream();
+		for (Topic topic : topics) {
+			for (Topic.TopicTerm term : topic.getTerms()) {
+				termList.add(term.getKeyword());
 			}
-		).map(
```
Comment (yuhai) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/messaging/InterestTermsChecker.java:86]: @W2jx, for the second loop, you may use termList.addAll(TransformUtil.transform(topic.getTerms(), term -> term.getKeyword())) to replace.

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java [hunk 3]
```diff
@@ -108,39 +109,42 @@ public long[] getSegmentsEntryIds(
...
+				});
+
+			List<Long> segmentsEntryIdList = new ArrayList<>();
+
+			for (SegmentsEntry segmentsEntry : segmentsEntries) {
+				segmentsEntryIdList.add(segmentsEntry.getSegmentsEntryId());
+			}
+
+			return ArrayUtil.toLongArray(segmentsEntryIdList);
 		}
```
Comment (yuhai) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java:143]: @W2jx, for segmentsEntryIdList part, please use TransformUtil.transformToArray.

**Comment by W2jx** [comment]
Hi @yuhai , I have fixed them.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:tango

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
no conflicts(1/3 8:56)

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/messaging/InterestTermsChecker.java [hunk 4]
```diff
@@ -75,19 +76,15 @@ public void checkInterestTerms(long companyId, String userId) {
...
-		Stream<Topic> stream = topics.stream();
+		List<String> termList = new ArrayList<>();
 
-		String[] terms = stream.flatMap(
-			topic -> {
-				List<Topic.TopicTerm> topicTerms = topic.getTerms();
+		for (Topic topic : topics) {
+			termList.addAll(
+				TransformUtil.transform(
+					topic.getTerms(), term -> term.getKeyword()));
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/messaging/InterestTermsChecker.java:84]: @W2jx You can use method reference here.

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java [hunk 5]
```diff
@@ -70,15 +71,12 @@ public long[] getSegmentsEntryClassPKs(
...
-				segmentsEntryId, start, end, null);
+				segmentsEntryId, start, end, null),
+			segmentsEntryRel -> segmentsEntryRel.getClassPK());
 
-		Stream<SegmentsEntryRel> stream = segmentsEntryRels.stream();
-
-		return stream.mapToLong(
-			SegmentsEntryRel::getClassPK
-		).toArray();
+		return ArrayUtil.toLongArray(classPKs);
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java:79]: @W2jx Please inline this list.

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java [hunk 6]
```diff
@@ -108,39 +106,36 @@ public long[] getSegmentsEntryIds(
...
 		if (GetterUtil.getBoolean(context.get(Context.SIGNED_IN))) {
-			List<SegmentsEntry> segmentsEntries =
+			List<SegmentsEntry> segmentsEntries = ListUtil.filter(
 				_segmentsEntryLocalService.getSegmentsEntries(
 					groupId, true,
 					SegmentsEntryConstants.SOURCE_ASAH_FARO_BACKEND, className,
-					QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
-
-			if (segmentsEntries.isEmpty()) {
-				return new long[0];
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java:118]: @W2jx The original logic has a shortcut when the list obtained from the local service is empty. Please try to keep it.

File: modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java [hunk 7]
```diff
@@ -108,39 +106,36 @@ public long[] getSegmentsEntryIds(
...
-			).mapToLong(
-				SegmentsEntry::getSegmentsEntryId
-			).toArray();
+				});
+
+			List<Long> segmentsEntryIdList = new ArrayList<>(
+				segmentsEntries.size());
+
+			for (SegmentsEntry segmentsEntry : segmentsEntries) {
+				segmentsEntryIdList.add(segmentsEntry.getSegmentsEntryId());
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/provider/AsahSegmentsEntryProvider.java:135]: @W2jx This loop can be replaced with a single call to `TransformUtil`.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:tango

**Comment by W2jx** [comment]
Hi @dantewang , I have modified them according to the comments.

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/2917

**Pull request closed without merging**